### PR TITLE
[FLINK-32208] Remove dependency on flink-shaded from flink-connector-aws 

### PR DESCRIPTION
**Pull request by dannycranmer**
- Remove the dependency on [flink-shaded](https://issues.apache.org/jira/browse/FLINK-shaded) for [flink-connector-aws](https://github.com/apache/flink-connector-aws)
- Remove Guava dependency from DDB connector
- Update NOTICE files

This is an update to https://github.com/apache/flink-connector-aws/pull/75

### CODE REVIEW

File: flink-connector-dynamodb/pom.xml [hunk 1]
```diff
@@ -92,6 +92,12 @@ under the License.
             <optional>true</optional>
         </dependency>
 
+        <!-- Other third-party dependencies -->
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+        </dependency>
+
```
Comment (dannycranmer) [inline flink-connector-dynamodb/pom.xml:100]: I think we can remove this dependency?

**Review by hlteoh37 — APPROVED** [review]
LGTM

File: flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/table/DynamoDbDynamicSinkFactory.java [hunk 2]
```diff
@@ -72,6 +70,10 @@ public String factoryIdentifier() {
 
     @Override
     public Set<ConfigOption<?>> requiredOptions() {
-        return ImmutableSet.of(TABLE_NAME, AWS_REGION);
+        final Set<ConfigOption<?>> requiredOptions = new HashSet<>();
+        requiredOptions.add(TABLE_NAME);
+        requiredOptions.add(AWS_REGION);
+
+        return requiredOptions;
```
Comment (z3d1k) [inline flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/table/DynamoDbDynamicSinkFactory.java:77]: Is this set available for external modification? If yes - `Collections.unmodifiableSet(requiredOptions)` can be used to preserve semantics of immutable collection.

File: flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSinkWriterTest.java [hunk 3]
```diff
@@ -68,17 +69,19 @@ public class DynamoDbSinkWriterTest {
...
-                        sinkPutRequest(item("pk", "1")),
-                        sinkPutRequest(item("pk", "2")),
-                        sinkDeleteRequest(item("pk", "3")),
-                        sinkDeleteRequest(item("pk", "4")));
+                Stream.of(
+                                sinkPutRequest(item("pk", "1")),
+                                sinkPutRequest(item("pk", "2")),
+                                sinkDeleteRequest(item("pk", "3")),
+                                sinkDeleteRequest(item("pk", "4")))
+                        .collect(toList());
```
[the remark this answers is not in the document]
Comment (z3d1k) [reply flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSinkWriterTest.java:77]: or use `.collect(toUnmodifiableList())` to preserve immutable semantics

[the remark this answers is not in the document]
Comment (dannycranmer) [reply flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSinkWriterTest.java:77]: I forgot about `Arrays.asList()` thanks

File: flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/table/DynamoDbDynamicSinkFactory.java (hunk 2 above)
Comment (dannycranmer) [reply flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/table/DynamoDbDynamicSinkFactory.java:77]: This is only used internally for Table API anyway

**Review by z3d1k — APPROVED** [review]
LGTM

**Pull request merged**